lib/xlsxtream/worksheet.rb

```diff
@@ -8,11 +8,25 @@ def initialize(io, options = {})
       @io = io
       @rownum = 1
       @options = options
+      @sheetdata_written = false
 
       write_header
     end
 
+    # If you want to specify custom column widths, do so using this
+    # method. See Xlsstream::Columns#initialize for parameter details.
+    # This MUST be called before #add_row or #<<.
+    #
+    def add_columns(column_options_array)
```

Comment:
I think the name `add_columns` is a bit confusing. Maybe `set_columns` or `configure_columns` would be better? 

What do you think?

Reply:
Or if the order *has* to be before the sheet data and we configure it through the worksheet initializer I'd probably just call it `write_columns` and make it private.

Reply:
OK, I'll have a think about that. I did originally intend that 'add columns' might be more flexible but it seems (from my limited understanding of the underlying XML format) to be quite constrained in terms of when it is set up, so a change of name would make sense.